src/types/presets.ts

```diff
@@ -52,6 +55,38 @@ export interface VercelBuildConfigV3 {
   }[];
 }
 
+interface FirebaseBaseOptions {
+  gen: 1 | 2;
+  /**
+   * Firebase functions node runtime version.
+   * @see https://cloud.google.com/functions/docs/concepts/nodejs-runtime
+   */
+  nodeVersion?: "20" | "18" | "16" | "14" | "12" | "10";
```

Comment:
To go with @pi0 comment [here](https://github.com/unjs/nitro/pull/1500/files/a0b602dc2662ffc09e149d85eeb9c00a417cdd83#r1278988243)

```suggestion
  nodeVersion?: "20" | "18" | "16";
```